polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerBackedByStoreFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.persistence;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.polaris.core.PolarisDiagnostics;
+import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.core.persistence.bootstrap.RootCredentialsSet;
+import org.apache.polaris.core.persistence.dao.entity.BaseResult;
+import org.apache.polaris.core.persistence.transactional.TransactionalPersistence;
+
+public abstract class LocalPolarisMetaStoreManagerBackedByStoreFactory<StoreType>
+    extends LocalPolarisMetaStoreManagerFactory {
+
+  final Map<String, StoreType> backingStoreMap = new HashMap<>();
+
+  protected LocalPolarisMetaStoreManagerBackedByStoreFactory(
+      @Nonnull PolarisDiagnostics diagnostics) {
+    super(diagnostics);
+  }
+
+  protected abstract StoreType createBackingStore(@Nonnull PolarisDiagnostics diagnostics);
+
+  protected abstract TransactionalPersistence createMetaStoreSession(
+      @Nonnull StoreType store,
+      @Nonnull RealmContext realmContext,
+      @Nullable RootCredentialsSet rootCredentialsSet,
+      @Nonnull PolarisDiagnostics diagnostics);
+
+  @Override
+  protected void initializeForRealm(
+      RealmContext realmContext, RootCredentialsSet rootCredentialsSet) {
+    final StoreType backingStore = createBackingStore(diagnostics);
+    sessionSupplierMap.put(
```

Comment:
I'm -1 on exposing mutable internal state to sub-classes.

What do we achieve by this refactoring? The end result is more confusing code, IMHO.

Reply:
If my old comment about extending `LocalPolarisMetaStoreManagerFactory` prompted this change, I'd prefer to go with the old state of this PR (not extending `LocalPolarisMetaStoreManagerFactory`) and then refactor later. Sorry about the confusion :sweat_smile:

Reply:
maybe i can break this into a function call which does this behind without exposing the state may be something like `cleanUpXXX` ? The whole intention was to re:use let me change that :)

Reply:
reverted the whole refactor, will pick it up after this gets in added TODO